sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/ProactiveOpenConnectionsProcessor.java

```diff
@@ -200,8 +205,25 @@ public void recordOpenConnectionsAndInitCachesStarted(List<CosmosContainerIdenti
             }
         }
     }
+    
+    public void recordCollectionRidsAndUrisUnderOpenConnectionsAndInitCaches(String collectionRid, List<String> addressUrisAsString) {
+        this.collectionRidsAndUrisUnderOpenConnectionAndInitCaches.putIfAbsent(collectionRid, new HashSet<>());
```

Comment:
This can be simplified (and made more efficient) by using ConcurrentHashMap.compute - the input would be the key and the current value (null if absent)) - so, something like

this.collectionRidsAndUrisUnderOpenConnectionAndInitCaches.compute(collectionRid, (ignore, urisAsString) -> {
    this.addressUrisUnderOpenConnectionsAndInitCaches.addAll(addressUrisAsString);
    
    if (urisAsString == null) {
        return new HashSet(addressUrisAsString);
    }

urisAsString.addAll(addressUrisAsString);
return urisAsString;
}